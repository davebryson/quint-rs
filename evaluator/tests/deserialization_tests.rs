use quint_evaluator::ir::QuintOutput;
use std::{env, fs::File};

#[test]
fn simple() {
    let file = File::open("fixtures/simple.json").unwrap();
    let parsed: QuintOutput = serde_json::from_reader(file).unwrap();
    insta::assert_yaml_snapshot!(parsed);
}

#[test]
fn tictactoe() {
    let file = File::open("fixtures/tictactoe.json").unwrap();
    let parsed: QuintOutput = serde_json::from_reader(file).unwrap();
    insta::assert_yaml_snapshot!(parsed);
}

#[test]
fn ewd840() {
    let file = File::open("fixtures/ewd840.json").unwrap();
    let parsed: QuintOutput = serde_json::from_reader(file).unwrap();
    insta::assert_yaml_snapshot!(parsed);
}

#[test]
fn ewd426() {
    let file = File::open("fixtures/ewd426.json").unwrap();
    let parsed: QuintOutput = serde_json::from_reader(file).unwrap();
    insta::assert_yaml_snapshot!(parsed);
}

#[test]
fn jmt() {
    let file = File::open("fixtures/jmt.json").unwrap();
    let parsed: QuintOutput = serde_json::from_reader(file).unwrap();
    insta::assert_yaml_snapshot!(parsed);
}
