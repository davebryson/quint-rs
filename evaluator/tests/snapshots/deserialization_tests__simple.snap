---
source: tests/deserialization_tests.rs
expression: parsed
---
modules:
  - name: test
    declarations:
      - kind: def
        id: 6
        name: f
        qualifier: puredef
        expr:
          kind: lambda
          id: 5
          params:
            - id: 1
              name: x_5
          expr:
            kind: app
            id: 4
            opcode: iadd
            args:
              - kind: name
                id: 2
                name: x_5
              - kind: int
                id: 3
                value: 1
        importedFrom: ~
        namespaces: ~
        depth: ~
      - kind: def
        id: 9
        name: a
        qualifier: pureval
        expr:
          kind: app
          id: 8
          opcode: f
          args:
            - kind: int
              id: 7
              value: 1
        importedFrom: ~
        namespaces: ~
        depth: ~
      - kind: def
        id: 12
        name: "q::init"
        qualifier: action
        expr:
          kind: bool
          id: 11
          value: true
        importedFrom: ~
        namespaces: ~
        depth: ~
      - kind: def
        id: 14
        name: "q::step"
        qualifier: action
        expr:
          kind: bool
          id: 13
          value: true
        importedFrom: ~
        namespaces: ~
        depth: ~
table:
  2:
    id: 1
    name: x_5
  6:
    kind: def
    id: 6
    name: f
    qualifier: puredef
    expr:
      kind: lambda
      id: 5
      params:
        - id: 1
          name: x_5
      expr:
        kind: app
        id: 4
        opcode: iadd
        args:
          - kind: name
            id: 2
            name: x_5
          - kind: int
            id: 3
            value: 1
    importedFrom: ~
    namespaces: ~
    depth: 0
  8:
    kind: def
    id: 6
    name: f
    qualifier: puredef
    expr:
      kind: lambda
      id: 5
      params:
        - id: 1
          name: x_5
      expr:
        kind: app
        id: 4
        opcode: iadd
        args:
          - kind: name
            id: 2
            name: x_5
          - kind: int
            id: 3
            value: 1
    importedFrom: ~
    namespaces: ~
    depth: 0
  9:
    kind: def
    id: 9
    name: a
    qualifier: pureval
    expr:
      kind: app
      id: 8
      opcode: f
      args:
        - kind: int
          id: 7
          value: 1
    importedFrom: ~
    namespaces: ~
    depth: 0
  12:
    kind: def
    id: 12
    name: "q::init"
    qualifier: action
    expr:
      kind: bool
      id: 11
      value: true
    importedFrom: ~
    namespaces: ~
    depth: 0
  14:
    kind: def
    id: 14
    name: "q::step"
    qualifier: action
    expr:
      kind: bool
      id: 13
      value: true
    importedFrom: ~
    namespaces: ~
    depth: 0
main: test
